src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -557,7 +557,11 @@ def _init_monitored_items(self):
 
         # make sure the metric_to_watch is an exact match
         metric_titles = self.loss_logging_items_names + get_metrics_titles(self.valid_metrics)
-        metric_to_watch_idx = fuzzy_idx_in_list(self.metric_to_watch, metric_titles)
+        try:
+            metric_to_watch_idx = fuzzy_idx_in_list(self.metric_to_watch, metric_titles)
+        except IndexError:
+            raise ValueError(f"No match found for `metric_to_watch={self.metric_to_watch}`. It should be one of {metric_titles}.")
```

Comment:
Great improvement!
A small comment regarding suggestion text - the formulation `It should be one of` may sound a little bit opaque to user. Maybe `Available metrics to monitor are: {metric_titles}` ?

Reply:
Good idea, changed in [ba708e2](https://github.com/Deci-AI/super-gradients/pull/1437/commits/ba708e259ad8745825a2fff50d10d4e0e1cadd4c)